docs/Localization.md

```diff
@@ -0,0 +1,83 @@
+The structure of the localization JSON files are key-value pairs, where:
+- The key is the reference to an element of `template.json` to be localized. If the element is child, the full path using `/` delimiter should be given, example: `symbols/Framework/choices/netstandard2.1/description`.
+- The value is the localization of the element given in the key. 
+
+Examples of localization files can be found [here](https://github.com/dotnet/templating/tree/main/template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ClassLibrary-CSharp/.template.config/localize).
+
+These files are be parsed by the template engine when loading information about the template. The Template Engine API returns information for localizable properties in language matching the current UI culture (if localization is available) transparently, without explicit user action.
+
+### Post action localization
+Ensure that all post actions in `template.json` have an `id` property prior setting up the localization. The `id` should be unique within the template. 
+Without the `id` property the localization files cannot be created.
+In case the post action uses more than one `manualInstructions`, `id` should be also added for each manual instruction.
+
+## Automatic generation of localization files
+
+The localization files can be generated automatically as part of a build using the `LocalizeTemplates` MSBuild task from the [`Microsoft.TemplateEngine.Tasks`](https://www.nuget.org/packages/Microsoft.TemplateEngine.Tasks) package.
+The task is meant to be used in template package project and will create the JSON files after the project is built.
+The task supports the configuration using following properties:
+- `LocalizableTemplatesPath` - the folder containing templates to be localized. Default value: `.`.
+- `TemplateLanguages` - the languages to create files for. Default value: `cs;de;en;es;fr;it;ja;ko;pl;pt-BR;ru;tr;zh-Hans;zh-Hant`.
+
+
+Example of usage:
+```xml
+<PropertyGroup>
+   <LocalizeTemplates>true</LocalizeTemplates>
+   <TemplateLanguages>en;de</TemplateLanguages>
+</PropertyGroup>
+
+<ItemGroup>
+   <PackageReference Include="Microsoft.TemplateEngine.Tasks" Version="1.0.0.0" PrivateAssets="all" IsImplicitlyDefined="true" />
+</ItemGroup>
+```
+
+This example will generate localization files for English and German languages for all templates in the package.
+
+## 1st party templates localization
+
+Prerequisite: the repo should be onboarded for OneLocBuild to localize the template content.
+See the [link](https://aka.ms/AllAboutLoc) for more details.
+
+### Repo is using arcade
+
+`Microsoft.TemplateEngine.Tasks` is included to `Microsoft.DotNet.Arcade.Sdk`.
+To start localizing the templates, enable the following property in template package project: <UsingToolTemplateLocalizer>true</UsingToolTemplateLocalizer>. This enables generating localization JSON files on the build.
```

Comment:
The property is not visible in markdown renderer. Hopefully this will fix it:
```suggestion
To start localizing the templates, enable the following property in template package project: `<UsingToolTemplateLocalizer>true</UsingToolTemplateLocalizer>`. This enables generating localization JSON files on the build.
```